src/yang/ietf-bgp-common.yang

```diff
@@ -340,6 +345,82 @@ submodule ietf-bgp-common {
          of an interface.";
     }
 
+    container secure-session {
+      must "enabled = 'false' or " +
+           "(enabled = 'true' and options)" {
+        error-message "When secure-session is enabled " +
+          "secure-session options MUST be set";
+      }
+      description
+        "Container for describing how a particular BGP session
+         is to be secured.";
+
+      leaf enabled {
+        type boolean;
+        default "false";
+        description
+          "When set to true, session is secured with the
+           configured options.";
+      }
+
+      container options {
+        description
+          "Options for securing the BGP session.";
+        choice option {
+          case ao {
+            leaf ao-keychain {
+              type key-chain:key-chain-ref;
+              description
+                "Reference to the key chain that will be used by this
+                 model. Applicable for TCP-AO and TCP-MD5 only";
+              reference
+                "RFC 8177: YANG Key Chain.";
+            }
+            description
+              "Uses TCP-AO to secure the session. Parameters for
+               those are defined as a grouping in the TCP YANG
+               model.";
+            reference
+              "RFC 5925 - The TCP Authentication Option.";
```

Comment:
While you are here, can you fix the format of this reference statement.

Reply:
Fixed in this file.